app/controllers/provider/admin/user/personal_details_controller.rb

```diff
@@ -34,6 +35,7 @@ def current_password_verification
 
     unless current_user.authenticated?(user_params[:current_password])
       flash.now[:error] = 'Current password is incorrect.'
+      AuditLogService.call("User tried to change password, but failed due to incorrect current password: #{current_user.id}/#{current_user.username}") if user_params[:password].present?
```

Comment:
why `user_params[:password].present?` ?

Reply:
should we rather check the future password fields instead? Honest question. Not sure that would be better.

Reply:
To me it seems like if `password` is set, then user tried to change password, and used wrong password or no password (which i technically also a wrong password).

If only current password was set but not new password, then user didn't try to change password. They just added noise.

Perhaps not a big deal but the intention IMO is shown by filling in the new password.

Reply:
I think we agree, and the code does what you say. Don't you think so?

Reply:
My complained was actually about
```
unless current_user.authenticated?(user_params[:current_password])
```

Why flash a message if user never supplied `:password` parameter? They didn't try to change password so they don't care that it didn't happen.

Reply:
Because if they don't provide `current_password` they can't change anything, not only the `password` field.

Reply:
got it, now it makes sense